.github/workflows/ci.yml

```diff
@@ -31,6 +31,56 @@ jobs:
       - name: Execute Pre-Commit
         run: pre-commit run --show-diff-on-failure --color=always --all-files
 
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
```

Comment:
I would instead use the minimal supported version. Imagine, that someone uses syntax or API from 3.12 in our common code base. It would pass this check but fail for 3.10 user.

Reply:
Agreed, although this is not introduced in this PR - we have been checking against 3.11 and more recently 3.12 via pre-commit. 

I'll make this my next PR 👍